dask/dataframe/io/parquet.py

```diff
@@ -74,6 +74,11 @@ def _read_fastparquet(fs, paths, myopen, columns=None, filters=None,
     elif len(minmax) > 1:
         if index:
             index_col = index
+        elif index is None:
+            # attempt auto-index
+            index = pf._get_index()
+            if index:
+                index_col = index
         elif 'index' in minmax:
             index_col = 'index'
```

Comment:
Chiming in here (though I haven't followed things closely).  I agree that there are certainly cases where this can result in confusing behavior.  Given the utility of the feature though I'm personally inclined to wait until a user issue arises before dealing with this.

Also, I wasn't aware that the pandas index isn't data.

@wesm is this a roadblock for your development in any way?  Or is this just a word of caution?